### ISSUE TITLE
abstract may lead to bad encodings when written out

### ISSUE THREAD

**Issue by kmccurley** [IACR/latex#150]
I don't have a repeatable example for this, but apparently this is a bug in using `fancyvrb` `\VerbatimOut`

**Comment by kmccurley** [comment]
I'm confused. This works on my machine when I compile it with pdflatex.

**Comment by kmccurley** [comment]
Can we at least track down what the issue was? From reading section 8 of https://ftp.snt.utwente.nl/pub/software/tex/macros/latex/contrib/fancyvrb/doc/fancyvrb-doc.pdf, it sounds like this is caused by T1-encoded characters, perhaps because it uses \write. This makes it sound like the problem we had in the server where the .log file produced by pdflatex can include multiple encodings. Maybe the issue is the version of fancyvrb that you are using.

**Comment by jwbos** [comment]
I am using:
\def\fileversion{4.1b}
\def\filedate{2021/12/21}
of fancyvrb.

**Comment by kmccurley** [comment]
I have changed to using texlive2023, which has version 4.5a of fancyvrb. This may be relevant to why I don't see the bug but you do. I could not find the source repository for fancyvrb, but there is a followon package fvextra: https://mirror.mwt.me/ctan/macros/latex/contrib/fvextra/fvextra.pdf This re-implements some of the internals of fancyvrb

**Comment by jwbos** [comment]
I tried a newer version of and got the same error. Somehow my input is utf8 but when pdflatex writes to a file it is not:

jwbos@jwbos-VirtualBox:~/latex/iacrcc$ file -i fail.tex 
fail.tex: text/x-tex; charset=utf-8

jwbos@jwbos-VirtualBox:~/latex/iacrcc$ file -i fail.abstract 
fail.abstract: text/plain; charset=iso-8859-1

**Comment by jwbos** [comment]
Kevin prodived this interesting example:

```
\documentclass{iacrcc}                                                                               
\usepackage[T1]{fontenc}                                                                             
\title{This is the title}
\begin{document}                                                                                     
\begin{abstract}No line at beginning.                                                                
  This mixes encodings with Ð and f\"ur and \th and a comment% but also a comment                    
and another line                                                                                     
\end{abstract}                                                                                       
This is not in the abstract.                                                                         
\end{document}
```

Using fancyvrb this is "detected" (providing the text immediately after the "\begin{abstract}") and throws an error:

```
FancyVerb Error:
Extraneous input `No line at beginning.' between \begin{abstract}[<key=value>
] and line end
.
\FV@Error ... {FancyVerb Error:
\space \space #1
}
```

When a newline is inserted then it still does not compile with an
```
! Undefined control sequence.
````
This seems to be happening because \jobname.abstract is stored using charset=us-ascii.

When using filecontents (part of LaTeX's kernel) then this is not detected.
The line "No line at beginning" is displayed *before* the abstract environment and is not part of the .abstract file.
(And everything compiles without an error). 
The charset of \jobname.abstract is utf-8.

Going forward with filecontents and trying to detect improper usage seems to be the solution?

**Comment by jwbos** [comment]
Some further investigation. I can patch fancyvrb following: https://tex.stackexchange.com/questions/596323/unicode-characters-within-verbatimout-of-fancyvrb
This solves the writing of unicode characters.

Maybe sticking with verbatimout seems the best way forward? Let's discuss in our weekly call.

**Comment by jwbos** [comment]
Fixed in 69f17c99b9ecc757f05e2cb9be3a539ee46e1344.
